docs/source/re_manager_api.rst

```diff
@@ -518,14 +519,75 @@ Returns       **success**: *boolean*
+
+**'queue_item_add_batch'**
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+============  =========================================================================================
+Method        **'queue_item_add_batch'**
+------------  -----------------------------------------------------------------------------------------
+Description   Add a batch of items to the back of the queue. The batch may consist of any number
+              of supported items (see **queue_item_add** method). The batch is treated as a single
+              unit. Each item in the queue must successfully pass validation before any items are added
+              to the queue. If any item fails validation, the whole batch is rejected. In case the
+              batch is rejected, the function returns the detailed report for each item, including
+              *success* status indicating if the item passed validation and error message in case
+              validation failed.
+------------  -----------------------------------------------------------------------------------------
+Parameters    **items**: *list*
+                  the list containing a batch of items. Each element is a dictionary representing
+                  a valid item. The dictionary must contain a key-value pair with the key representing
+                  one of the supported item types (*'plan'* or *'instruction'*) and the value being
+                  a dictionary of item parameters. An empty item list will also be accepted.
+
+              **user_group**: *str*
+                  the name of the user group (e.g. 'admin').
+
+              **user**: *str*
+                  the name of the user (e.g. 'John Doe'). The name is included in the item metadata
+                  and may be used to identify the user who added the item to the queue. It is not
+                  passed to the Run Engine or included in run metadata.
+------------  -----------------------------------------------------------------------------------------
+Returns       **success**: *boolean*
+                  indicates if the request was processed successfully. The request fails if any item
+                  fails validation. The queue is not expected to be modified if the request fails.
+                  If the parameter is *True*, then validation of all items succeeded: there is no
+                  need to verify *success* status for each item returned in *item_list* parameter.
+
+              **msg**: *str*
+                  error message in case of failure, empty string ('') otherwise.
+
+              **qsize**: *int* or *None*
+                  the number of items in the plan queue after processing of the request. The correct
+                  queue size may be returned even if the operation fails. In rare failing cases
+                  the parameter may return *None*.
+
+              **item_list** : *list*
+                  the list of processed items inserted item. Each item represents the dictionary
```

Comment:
```suggestion
                  the list of processed items. Each item represents the dictionary
```

Reply:
Thanks for careful reading of the documentation. I will implement changes in the next PR.